eZ/Publish/API/Repository/Tests/SearchServiceDeleteTranslationTest.php

```diff
@@ -0,0 +1,132 @@
+     * @throws \eZ\Publish\API\Repository\Exceptions\ContentValidationException
+     * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
+     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException
+     */
+    protected function createTestContentWithLanguages(array $languages): Content
+    {
+        $repository = $this->getRepository();
+        $contentTypeService = $repository->getContentTypeService();
+        $contentService = $repository->getContentService();
+        $locationService = $repository->getLocationService();
+
+        $contentTypeArticle = $contentTypeService->loadContentTypeByIdentifier('article');
+        $contentCreateStructArticle = $contentService->newContentCreateStruct(
+            $contentTypeArticle,
+            'eng-GB'
+        );
+
+        foreach ($languages as $langCode => $title) {
+            $contentCreateStructArticle->setField('title', $title, $langCode);
+            $contentCreateStructArticle->setField(
+                'intro',
+                '<?xml version="1.0" encoding="UTF-8"?>
+<section xmlns="http://docbook.org/ns/docbook" version="5.0-variant ezpublish-1.0">
+  <para>' . $title . '</para>
+</section>',
+                $langCode
+            );
+        }
+
+        $locationCreateStructArticle = $locationService->newLocationCreateStruct(2);
+        $draftArticle = $contentService->createContent(
+            $contentCreateStructArticle,
+            [$locationCreateStructArticle]
+        );
+        $content = $contentService->publishVersion($draftArticle->getVersionInfo());
+        $this->refreshSearch($repository);
+
+        return $content;
+    }
+
+    /**
+     * @return \eZ\Publish\API\Repository\Values\Content\Search\SearchResult
+     * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException
+     */
+    protected function findContent(string $text, string $languageCode): SearchResult
+    {
+        $repository = $this->getRepository();
+        $searchService = $repository->getSearchService();
+        $query = new Query();
+        $query->query = new Criterion\FullText($text);
+        $query->limit = 0;
+        $languageFilter = [
+            'languages' => [$languageCode],
+            'useAlwaysAvailable' => true,
+            'excludeTranslationsFromAlwaysAvailable' => false,
+        ];
+
+        return $searchService->findContent($query, $languageFilter);
+    }
+
+    /**
+     * @throws \eZ\Publish\API\Repository\Exceptions\BadStateException
+     * @throws \eZ\Publish\API\Repository\Exceptions\ContentFieldValidationException
+     * @throws \eZ\Publish\API\Repository\Exceptions\ContentValidationException
+     * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
+     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException
```

Comment:
AFAIR Some of those throw common ForbiddenException. Replace them with that please.